### ISSUE TITLE
Minor header file issues

### ISSUE THREAD

**Issue by wm4** [BYVoid/uchardet#6]
There are some minor issues in the public header file.

```C
#ifndef ___UCHARDET_H___
```

Identifiers starting with 2 _ are reserved by the system. Normal libraries cannot use such identifiers.

```C
typedef void * uchardet_t;
```

This isn't really idea. Using void* reduces type-safety, and the typedef obscures the real type of the handle. Also, ``_t`` suffixes are reserved for POSIX standard symbols. I suggest doing ``typedef struct uchardet uchardet;`` and using it as ``uchardet *ud`` in function parameters.

```C
uchardet_t uchardet_new();
```

The parameter list is empty - in C, this means the parameter list is not defined, and anything is allowed. clang e.g. warns with: ``uchardet/uchardet.h:52:1: warning: function declaration isn't a prototype``. It should be ``uchardet_t uchardet_new(void);``. This compiles in C++ too.

Also, this should probably define that it returns NULL on memory allocation failure.

**Comment by BYVoid** [comment]
Thanks.

**Comment by wm4** [comment]
Fixed with #8. The ``_t`` issue (clashing with POSIX) is unfortunately not fixable without a major API change. Thanks for merging my PR!
